airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -36,6 +36,12 @@ def __init__(
     ):
         self._timezone = datetime.timezone.utc
         self._interpolation = JinjaInterpolation()
+        if isinstance(start_time, str):
```

Comment:
In the above `__init__` signature, should we add `Union[InterpolatedString, str]`. Otherwise it gives the impression that we only ever accept InterpolatedStrings but in reality a str might come in hence why we need these lines